### ISSUE TITLE
Detect @abstractmethod used in non-abstract classes

### ISSUE THREAD

**Issue by eltoder** [python/mypy#14686]
Consider the following example:
```python
import abc

class C:
    @abc.abstractmethod
    def method(self) -> None:
        raise NotImplementedError
```
Currently mypy does not report any errors for this file. However, note that `C` does not inherit from `abc.ABC` and does not use `abc.ABCMeta` as its metaclass. This means that `@abstractmethod` is ignored at runtime and classes that don't implement `method` can still be instantiated. This is most likely a mistake that can lead to raising `NotImplementedError` at runtime. It will be helpful if mypy reports an error when `@abstractmethod` decorator is used in classes that do not inherit from `abc.ABC`.

This feature [is present in pytype](https://google.github.io/pytype/errors.html#ignored-abstractmethod).

**Comment by hauntsaninja** [comment]
mypy actually does the opposite as a feature. mypy will consider classes with an abstractmethod as abstract even without inheriting from ABC or having ABCMeta metaclass and type check them as such (so as long as you use a type checker you have the same guarantee). This is useful e.g. if adding ABC would cause metaclass conflicts.

This is a documented feature (see https://mypy.readthedocs.io/en/stable/class_basics.html#abstract-base-classes-and-multiple-inheritance), and I'm not seeing any similar requests on the issue tracker, so it's unlikely that we will change this.
